pkg/api/meta/meta_test.go

```diff
@@ -722,6 +729,57 @@ func TestTypeMetaSelfLinker(t *testing.T) {
 	}
 }
 
+type MyAPIObject2 struct {
+	unversioned.TypeMeta
+	v1.ObjectMeta
+}
+
+func getObjectMetaAndOwnerReferneces() (MyAPIObject2, []metatypes.OwnerReference) {
+	return MyAPIObject2{
+			ObjectMeta: v1.ObjectMeta{
+				OwnerReferences: []v1.OwnerReference{
+					{UID: "1"},
+					{UID: "2"},
+					{UID: "3"},
+					{UID: "4"},
+					{UID: "5"},
+				},
+			},
+		},
+		[]metatypes.OwnerReference{
+			{UID: "1"},
+			{UID: "2"},
+			{UID: "3"},
+			{UID: "4"},
+			{UID: "5"},
+		}
+}
+
+func TestGetOwnerReferences(t *testing.T) {
+	obj, expected := getObjectMetaAndOwnerReferneces()
+	accessor, err := meta.Accessor(&obj)
+	if err != nil {
+		t.Error(err)
+	}
+	references := accessor.GetOwnerReferences()
+	if !reflect.DeepEqual(references, expected) {
+		t.Errorf("expect %v, got %v", expected, references)
+	}
+}
+
+func TestSetOwnerReferences(t *testing.T) {
```

Comment:
These tests are not sufficient. You need to test every field. I recommend using the fuzz library, round trip, and reflect.DeepEqual.